core/admin.py

```diff
@@ -32,24 +48,58 @@ class Meta:
             'content': RedactorWidget(editor_options={'lang': 'en'})
         }
 
+
+class SponsorInline(SortableTabularInline):
+    model = Sponsor
+    extra = 1
+    verbose_name_plural = 'Sponsors'
+    sortable = 'position'
+    fields = ('name', 'logo', 'url', 'position')
+
+
+class CoachInline(admin.TabularInline):
+    model = Coach
+    extra = 1
+    verbose_name_plural = 'Coaches'
+    sortable = 'position'
+    fields = ('name', 'twitter_handle', 'url', 'photo')
+
+
 class EventPageContentAdmin(SortableModelAdmin):
     list_display = ('name', 'page', 'content', 'position', 'is_public')
-    list_filter = ('page','is_public')
+    list_filter = ('page', 'is_public')
+    search_fields = ('name', 'page__title', 'content', 'page__event__city',
+                     'page__event__country', 'page__event__name')
     form = EventPageContentForm
     sortable = 'position'
+    inlines = [
+        SponsorInline,
+        CoachInline
+    ]
 
     def get_queryset(self, request):
         qs = super(EventPageContentAdmin, self).queryset(request)
         if request.user.is_superuser:
             return qs
-        return qs.filter(page__event__team__in=[request.user,])
+        return qs.filter(page__event__team__in=[request.user])
 
     def get_form(self, request, obj=None, **kwargs):
         form = super(EventPageContentAdmin, self).get_form(request, obj, **kwargs)
         if not request.user.is_superuser:
-            form.base_fields['page'].queryset = EventPage.objects.filter(event__team__in=[request.user])
+            if 'page' in form.base_fields:
+                form.base_fields['page'].queryset = EventPage.objects.filter(
```

Comment:
I think you should use `form.fields` here and not `form.base_fields` because otherwise you're changing every single instance of that form ever.